encoding/marshal.go

```diff
@@ -17,10 +17,9 @@ type SiaMarshaler interface {
 	MarshalSia() []byte
 }
 
-// A SiaUnmarshaler can decode itself from a byte slice. If a decoding error
-// occurs, UnmarshalSia should panic.
+// A SiaUnmarshaler can decode itself from a byte slice.
 type SiaUnmarshaler interface {
-	UnmarshalSia([]byte)
+	UnmarshalSia([]byte) error
```

Comment:
also we need an int I think, so that the number of bytes used can be returned. UnmarshalSia and MarshalSia can't add a prefix that wouldn't be there otherwise.